cdrom/src/lib.rs

```diff
@@ -435,7 +435,11 @@ impl Sector {
     //
     // More information is in ECMA-130:
     // http://www.ecma-international.org/publications/standards/Ecma-130.htm
-    pub fn generate_subchannel(&self) -> Vec<u8> {
+    pub fn generate_subchannel(
+        &self,
+        protection: Option<bool>,
+        chosen_protection_type: Option<DiscProtection>,)
```

Comment:
We can simplify this a little bit using Rust's `Option` type. The `Option` type represents a value that *might* be there, or might not - so, in other words, we don't need a separate boolean to track if we want protection or not. If the protection is `None`, we know we don't want it; if it's `Some(type)`, we know both that we want it *and* what type it is, in the same variable!